src/util/config.cpp

```diff
@@ -269,9 +295,9 @@ std::string configt::this_architecture()
   return this_arch;
 }
 
-std::string configt::this_operating_system()
+const char *configt::this_operating_system()
```

Comment:
Why using const char* here instead of std::string?

Reply:
Because there is no need for dynamic allocation (it's really some constant) and it is implicitely convertible to std::string.